neuromation/api/config_factory.py

```diff
@@ -141,6 +150,32 @@ async def login_with_token(
         )
         self._save(config)
 
+    async def login_with_passed_config(
+        self,
+        config_data: Optional[str] = None,
+        *,
+        timeout: aiohttp.ClientTimeout = DEFAULT_TIMEOUT,
+    ) -> None:
+        if config_data is None:
+            try:
+                config_data = os.environ[PASS_CONFIG_ENV_NAME]
+            except KeyError:
+                raise ConfigError(
+                    f"Config env variable {PASS_CONFIG_ENV_NAME} " "is not present"
+                )
+        try:
+            data = json.loads(base64.b64decode(config_data).decode())
+            token = data["token"]
+            cluster = data["cluster"]
+            url = URL(data["url"])
+        except (ValueError, KeyError):
+            raise ConfigError(f"Data in passed config is malformed: {config_data}")
+        await self.login_with_token(token, url=url, timeout=timeout)
+        client = await self.get(timeout=timeout)
+
+        await client.config.switch_cluster(cluster)
+        await client.close()
```

Comment:
Even if the new code doesn't use the storage saved configuration for the login -- cleaning up the storage is still a good idea unless we stop pushing configuration files on the storage.
Please lookup if the config is present and clean it up -- as `steal_config_maybe()` did for the previous version.